src/test/java/com/rackspace/ceres/app/services/DataWriteServiceTest.java

```diff
@@ -115,12 +112,14 @@ void testSingle() {
       final String metricName = RandomStringUtils.randomAlphabetic(5);
       final String metricGroup = RandomStringUtils.randomAlphabetic(5);
       final String resource = RandomStringUtils.randomAlphabetic(5);
+      final String monitoring_system = RandomStringUtils.randomAlphanumeric(5);
       final Map<String, String> tags = Map.of(
           "os", "linux",
           "host", "h-1",
           "deployment", "prod",
           "metricGroup", metricGroup,
-          "resource", resource
+          "resource", resource,
+          "monitoring_system", monitoring_system
```

Comment:
while asserting timer count metric below you can assert new sensor tag added as part of latency. May be something like 
```
      Timer timer = meterRegistry.get("ingest.latency").timer();
      assertThat(timer.count()).isGreaterThanOrEqualTo(1L);
      assertThat(timer.getId().getTag("sensor_name")).isEqualTo(monitoring_system);
```

Reply:
asserting in one place is enough while. You could add assertion in DataWriteServiceTest#testSingle method.